staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -291,10 +291,41 @@ type CSIDriverSpec struct {
 	// "csi.storage.k8s.io/pod.name": pod.Name
 	// "csi.storage.k8s.io/pod.namespace": pod.Namespace
 	// "csi.storage.k8s.io/pod.uid": string(pod.UID)
+	// "csi.storage.k8s.io/ephemeral": "true" iff the volume is an ephemeral inline volumed
+	//                                 defined by a CSIVolumeSource, otherwise "false"
 	// +optional
 	PodInfoOnMount *bool `json:"podInfoOnMount,omitempty" protobuf:"bytes,2,opt,name=podInfoOnMount"`
+
+	// If set, Mode defines how this CSI volume driver can be used.
```

Comment:
style: use `mode` instead of `Mode` since the godoc is intended for people who are using the API via JSON, not Go code.

Reply:
After discussion in the review call, we (@msau42 @jsafrane and @smarterclayton) agreed this should probably be an array of enumerations, so `(DriverModes|SupportDriverModes|AllowedDriverModes) []CSIDriverMode` with enumeration values `Persistent`, `Ephemeral`.

DriverModes may be the best, please think about how a user reading this would think about it.

Reply:
I agree, turning this into an array makes sense.

Instead using the word "mode" (which is awfully overloaded), would it make sense turn this new field in a generic "Capabilities" array? That then opens the door for adding future enumeration values that having nothing to do with the current usage (volume "modes").

Reply:
Yes another option is one field per capability:
- `PersistentVolumes *bool` (default true)
- `EphemeralVolumes *bool` (default false)

Reply:
I actually like that last option best myself.

Reply:
I prefer putting these two into a separate array, because these two capabilities and the functionality that they impact are correlated very closely with each other.

Reply:
How would an enumeration where multiple can be used work in yaml?

PersistentVolumes and EphemeralVolumes would work for me.

Or:
Mode:
 - Persistent
 - Ephemeral

Where the array entries are strings and its existence in the list is true. Maybe thats what @smarterclayton was meaning?

Reply:
@kfox1111: yes, that's the proposal from https://github.com/kubernetes/kubernetes/pull/80568#discussion_r309775598.

Okay, as as the flags are meant to be semantically related, then let's go with that proposal and define it as:
```
VolumeModes:
- Persistent
- Ephemeral
```

If the list is empty or nil,  "Persistent" is added as default.

I chose this naming because IMHO it makes more sense to describe it in terms of "what does the driver support" instead of "in which mode does the driver operate", in particular when the driver switches on a volume-by-volume basis.